tests/integration/cocoon/test_cocoon_portfolio_groups.py

```diff
@@ -0,0 +1,324 @@
+
+        self.assertEqual(
+            json.loads(responses["portfolio_groups"]["errors"][0].body)["name"],
+            "PortfolioNotFound",
+        )
+
+    def test_duplicate_portfolio(self):
+
+        """
+        Description:
+        ------------
+        Here we test attempting to add two of the same portfolios to a portfolio group.
+        """
+
+        duplicate_df = pd.read_csv(
+            Path(__file__).parent.joinpath(
+                "data/port_group_tests/portfolio_group_duplicate_portfolio.csv"
+            )
+        )
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=duplicate_df,
+            mapping_required={
+                "code": "PortGroupCode",
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={"values.scope": "Scope", "values.code": "FundCode",},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
+            for error in responses["portfolio_groups"]["errors"]:
+                print(json.loads(error.body)["title"])
+
+        duplicate_df["unique_combinations_for_test"] = (
+            duplicate_df["PortGroupCode"]
+            + "-"
+            + duplicate_df["FundCode"]
+            + "-"
+            + duplicate_df["Scope"]
+        )
+
+        self.assertEqual(
+            len(
+                [
+                    port_group
+                    for nested_group in [
+                        port_group.portfolios
+                        for port_group in responses["portfolio_groups"]["success"]
+                    ]
+                    for port_group in nested_group
+                ]
+            ),
+            len(duplicate_df["unique_combinations_for_test"].unique()),
+        )
+
+    def test_port_group_update(self):
+
+        """
+        Description:
+        ------------
+        Here we test updating a portfolio group which already exists.
+        """
+
+        update_df = pd.read_csv(
+            Path(__file__).parent.joinpath(
+                "data/port_group_tests/portfolio_group_update.csv"
+            )
+        )
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=update_df,
+            mapping_required={
+                "code": "PortGroupCode",
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={"values.scope": "Scope", "values.code": "FundCode",},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
+            for error in responses["portfolio_groups"]["errors"]:
```

Comment:
Better practice here would be to check that the length of the list is greater than 0.

Why did you use .get here "responses.get("portfolio_groups").get("errors")" and then [] syntax a few lines later "responses["portfolio_groups"]["errors"]"

Reply:
As above.